backends/tfhe-cuda-backend/cuda/src/crypto/keyswitch.cu

```diff
@@ -40,11 +40,63 @@ void cuda_keyswitch_lwe_ciphertext_vector_64(
     void *lwe_output_indexes, void *lwe_array_in, void *lwe_input_indexes,
     void *ksk, uint32_t lwe_dimension_in, uint32_t lwe_dimension_out,
     uint32_t base_log, uint32_t level_count, uint32_t num_samples) {
-  cuda_keyswitch_lwe_ciphertext_vector(
+  host_keyswitch_lwe_ciphertext_vector(
       static_cast<cudaStream_t>(stream), gpu_index,
       static_cast<uint64_t *>(lwe_array_out),
       static_cast<uint64_t *>(lwe_output_indexes),
       static_cast<uint64_t *>(lwe_array_in),
       static_cast<uint64_t *>(lwe_input_indexes), static_cast<uint64_t *>(ksk),
       lwe_dimension_in, lwe_dimension_out, base_log, level_count, num_samples);
 }
+
+/* Perform functional packing keyswitch on a batch of 64 bits input LWE
+ * ciphertexts.
+ *
+ * - `v_stream` is a void pointer to the Cuda stream to be used in the kernel
+ * launch
+ * - `gpu_index` is the index of the GPU to be used in the kernel launch
+ * - `glwe_array_out`: output batch of keyswitched ciphertexts
+ * - `lwe_array_in`: input batch of num_samples LWE ciphertexts, containing
+ * lwe_dimension_in mask values + 1 body value
+ *  - `fp_ksk_array`: the functional packing keyswitch keys to be used in the
+ * operation
+ *  - `base log`: the log of the base used in the decomposition (should be the
+ * one used to create the ksk)
+ *  - `level_count`: the number of levels used in the decomposition (should be
+ * the one used to  create the fp_ksks).
+ *  - `number_of_input_lwe`: the number of inputs
+ *  - `number_of_keys`: the number of fp_ksks
+ *
+ * This function calls a wrapper to a device kernel that performs the functional
+ * packing keyswitch.
+ */
+void cuda_fp_keyswitch_lwe_to_glwe_64(void *stream, uint32_t gpu_index,
+                                      void *glwe_array_out, void *lwe_array_in,
+                                      void *fp_ksk_array,
+                                      uint32_t input_lwe_dimension,
+                                      uint32_t output_glwe_dimension,
+                                      uint32_t output_polynomial_size,
+                                      uint32_t base_log, uint32_t level_count) {
```

Comment:
Why is there an entry point for a list and one for a single ciphertext?  We could just call always the entry point for the list and give num_lwes = 1 no?

Reply:
Yes, we can. Since the packing keyswitch implementation for a list of LWEs depends on implementing it for a single ciphertexts I decided to leave the entry point for completeness. But yeah, we don't use it today and we can just use the other one with `num_lwes = 1`. I will remove it.